packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -210,6 +210,73 @@ export enum NodeKind {
 	Leaf,
 }
 
+/**
+ * The unique identifier of a field, used in the persisted form of the tree.
+ *
+ * @remarks
+ * If not explicitly set via the schema, this is the same as the schema's property key.
+ *
+ * Specifying a stored key that differs from the property key is particularly useful in refactoring scenarios.
+ * To update the developer-facing API, while maintaining backwards compatibility with existing SharedTree data,
+ * you can change the property key and specify the previous property key as the stored key.
+ *
+ * Notes:
+ *
+ * - Stored keys have no impact on standard JavaScript behavior, on tree nodes. For example, `Object.keys`
+ * will always return the property keys specified in the schema, ignoring any stored keys that differ from
+ * the property keys.
+ *
+ * - When specifying stored keys in an object schema, you must ensure that the final set of stored keys
+ * (accounting for those implicitly derived from property keys) contains no duplicates.
+ * This is validated at runtime.
+ *
+ * @example Refactoring code without breaking compatibility with existing data
+ *
+ * Consider some existing object schema:
+ *
+ * ```TypeScript
+ * class Point extends schemaFactory.object("Point", {
+ * 	xPosition: schemaFactory.number,
+ * 	yPosition: schemaFactory.number,
+ * 	zPosition: schemaFactory.optional(schemaFactory.number),
+ * });
+ * ```
+ *
+ * Developers using nodes of this type would access the the `xPosition` property as `point.xPosition`.
+ *
+ * We would like to refactor the schema to omit "Position" from the property keys, but application data has
+ * already been persisted using the original property keys. To maintain compatibility with existing data,
+ * we can refactor the schema as follows:
+ *
+ * ```TypeScript
+ * class Point extends schemaFactory.object("Point", {
+ * 	x: schemaFactory.required(schemaFactory.number, { key: "xPosition" }),
+ * 	y: schemaFactory.required(schemaFactory.number, { key: "yPosition" }),
+ * 	z: schemaFactory.optional(schemaFactory.number, { key: "zPosition" }),
+ * });
+ * ```
+ *
+ * Now, developers can access the `x` property as `point.x`, while existing data can still be collaborated on.
+ *
+ * @example
+ *
+ * @public
+ */
+export type StoredKey = string;
+
+/**
+ * Properties associated with a field.
```

Comment:
We made the choice to not include allowedTypes or the field kind here (which is something we could have done), so we might want to tweak the phrasing here. Also "properties" are generated from fields on objects (but specifically not with these keys), so to reduce risks of confusion I'm picking a different term:
```suggestion
 * Additional information to provide to a {@link FieldSchema}.
```